component/common/loki/wal/config.go

```diff
@@ -20,17 +21,37 @@ type Config struct {
 	//
 	// WAL support is a WIP. Do not enable in production setups until https://github.com/grafana/loki/issues/8197
 	// is finished.
-	Enabled bool `yaml:"enabled"`
+	Enabled bool `yaml:"enabled" river:"enabled,attr,optional"`
 
 	// Path where the WAL is written to.
-	Dir string `yaml:"dir"`
+	Dir string `yaml:"dir" river:"dir,attr,optional"`
```

Comment:
This is a bit of an anti-pattern for Flow I think; components are given directories unique to that component and they should use that for storage instead of having it be configurable.

Reply:
Ah you mean `opts.DataPath` right? Do you have an example of a component using it? Is that directory kept the same if one restarts the agent?

Reply:
> Ah you mean opts.DataPath right? Do you have an example of a component using it?

Yeah, for an example, look at this in prometheus.remote_write where the WAL is stored:

https://github.com/grafana/agent/blob/ba235197332fb071a126a3389ce12367ff020850/component/prometheus/remotewrite/remote_write.go#L75-L76

> Is that directory kept the same if one restarts the agent?

Depends on how the agent is configured. All the components get their data paths from the `--storage.path` flag; typically if the user wants it to persist they would set `--storage.path` to something which persists between restarts/reboots.

Reply:
Applied the same idea used in the prom remote write

Reply:
Ok, using a sub-dir under the component data dir.